cli/flox-activations/src/cli/set_env_dirs.rs

```diff
@@ -0,0 +1,97 @@
+use std::path::PathBuf;
+
+use anyhow::bail;
+use clap::Args;
+use log::debug;
+
+use super::{join_dir_list, separate_dir_list};
+
+#[derive(Debug, Args)]
+pub struct SetEnvDirsArgs {
+    #[arg(short, long, help = "The contents of the FLOX_ENV variable")]
+    pub flox_env: PathBuf,
+    #[arg(
+        short,
+        long,
+        help = "The contents of the FLOX_ENV_DIRS variable (which may be unset or empty)."
+    )]
+    pub env_dirs: String,
+    #[arg(short, long, help = "Which shell syntax to emit")]
+    pub shell: String,
+}
+
+impl SetEnvDirsArgs {
+    pub fn handle(&self) -> Result<(), anyhow::Error> {
+        let mut output = std::io::stdout();
+        self.handle_inner(&mut output)
+    }
+
+    fn handle_inner(&self, output: &mut impl std::io::Write) -> Result<(), anyhow::Error> {
+        debug!(
+            "Preparing to set FLOX_ENV_DIRS, FLOX_ENV={}, FLOX_ENV_DIRS={}",
+            self.flox_env.display(),
+            self.env_dirs
+        );
+        let mut existing_dirs = separate_dir_list(&self.env_dirs);
+        if existing_dirs.is_empty() || !existing_dirs.contains(&self.flox_env) {
+            existing_dirs.insert(0, self.flox_env.clone());
+        }
+        let joined = join_dir_list(existing_dirs);
+        let sourceable_commands = match self.shell.as_ref() {
+            "bash" => {
+                format!("export FLOX_ENV_DIRS=\"{joined}\";")
+            },
+            "zsh" => {
+                format!("export FLOX_ENV_DIRS=\"{joined}\";")
+            },
+            "fish" => {
+                format!("set -gx FLOX_ENV_DIRS \"{joined}\";")
+            },
+            "tcsh" => {
+                format!("setenv FLOX_ENV_DIRS \"{joined}\";")
+            },
+            other => {
+                bail!("invalid shell: {other}")
+            },
+        };
+        output.write_all(sourceable_commands.as_bytes())?;
+        debug!("Set FLOX_ENV_DIRS, FLOX_ENV_DIRS={}", joined);
```

Comment:
nit
```suggestion
        debug!("Set FLOX_ENV_DIRS={}", joined);
```

Reply:
This is poor man's structured logging without `tracing`, so I'll leave it how it is.